broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -737,6 +737,7 @@ public void onFailure() {
           if (failureListener != null) {
             failureListener.onFailure();
           }
+          localBroker.setPartitionUnhealthy(partitionId);
```

Comment:
I think that for this (and the other cases below), the changes to the localBroker data also need to be published. For example, this is also done by the `TopologyManagerImpl` when the partition role changes (i.e. becoming leader or follower). See `TopologyManagerImpl#L218` to see how this is done.

@deepthidevaki WDYT, is it necessary here to publish these changes explicitly?

Reply:
You are right @korthout. It needs to be written into `atomix.getMembershipService().getLocalMember().properties()` for it to be picked up in the next gossip propagation.

Reply:
Also note that `localBroker` is not a thread safe object. It can be concurrently updated by TopologyManagerImpl. So it is not safe to update it directly from here. May be you can delegate it to TopologyManagerImpl.